lib/src/puppetdb-cli.cc

```diff
@@ -107,7 +121,85 @@ PuppetDBConn::parseServerUrls(const json::JsonContainer& config) {
     }
 }
 
-size_t write_data(void *ptr, size_t size, size_t nmemb, FILE *stream) {
+class JsonQueue {
```

Comment:
The name `JsonQueue` doesn't make sense to me here. Looks like a `SynchronizedCharQueue` to me, and I'd be tempted to make it a `SynchronizedQueue<T>` if that makes sense.